src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -564,35 +586,37 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
           ) : null}
         </Row>
       </AnimatedBox>
-      <InfiniteScroll
-        next={() => loadNext(ASSET_PAGE_SIZE)}
-        hasMore={hasNext}
-        loader={hasNext && hasNfts ? loadingAssets() : null}
-        dataLength={collectionAssets?.length ?? 0}
-        style={{ overflow: 'unset' }}
-        className={hasNfts || isLoadingNext ? styles.assetList : undefined}
-      >
-        {hasNfts ? (
-          assets
-        ) : collectionAssets?.length === 0 ? (
-          <Center width="full" color="textSecondary" textAlign="center" style={{ height: '60vh' }}>
-            <EmptyCollectionWrapper>
-              <p className={headlineMedium}>No NFTS found</p>
-              <Box
-                onClick={reset}
-                type="button"
-                className={clsx(bodySmall, buttonTextMedium)}
-                color="blue"
-                cursor="pointer"
-              >
-                <ViewFullCollection>View full collection</ViewFullCollection>
-              </Box>
-            </EmptyCollectionWrapper>
-          </Center>
-        ) : (
-          <CollectionNftsLoading />
-        )}
-      </InfiniteScroll>
+      <InfiniteScrollWrapper>
+        <InfiniteScroll
+          next={handleNextPageLoad}
+          hasMore={hasNext}
+          loader={Boolean(hasNext && hasNfts) && <LoadingAssets />}
```

Comment:
how could this ever not be a boolean?

Reply:
couldn't--I just hate the pattern of unwrapped compound bool statements in templates because SOMETIMES IT BITES

Reply:
I don't let myself get comfortable with unwrapped ones so I don't accidentally ship a bug